File: EdgePartition/CLUGP/src/ustc.ddl/thread/ClusterGameTask.java
package ustc.ddl.thread;

import ustc.ddl.cluster.StreamCluster;
import ustc.ddl.game.ClusterPackGame;
import ustc.ddl.properties.GlobalConfig;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.concurrent.Callable;

public class ClusterGameTask implements Callable<ClusterPackGame> {

    private final StreamCluster streamCluster;
    private final List<Integer> cluster;

    public ClusterGameTask(StreamCluster streamCluster, int taskId) {
        this.streamCluster = streamCluster;

        int batchSize = GlobalConfig.getBatchSize();
        List<Integer> clusterList = streamCluster.getClusterList();
        int begin = batchSize * taskId;
        int end = Math.min(batchSize * (taskId + 1), clusterList.size());

        cluster = clusterList.subList(begin, end);
    }

    @Override
    public ClusterPackGame call() throws Exception {
        ClusterPackGame clusterPackGame = new ClusterPackGame(streamCluster, cluster);
        clusterPackGame.initGame();
        clusterPackGame.startGame();
        return clusterPackGame;
    }
}

File: EdgePartition/CLUGP/C++/ClusterGameTask.cpp
#include "ClusterGameTask.h"


ClusterGameTask::ClusterGameTask(GlobalConfig& config,std::shared_ptr<StreamCluster> streamCluster, int taskId) : streamCluster(streamCluster) {
    int batchSize = config.getBatchSize();
    std::vector<int> clusterList = streamCluster->getClusterList();
    int begin = batchSize * taskId;
    int end = std::min(batchSize * (taskId + 1), static_cast<int>(clusterList.size()));

    cluster.assign(clusterList.begin() + begin, clusterList.begin() + end);
}

std::shared_ptr<ClusterPackGame> ClusterGameTask::call(GlobalConfig& config) {
    std::shared_ptr<ClusterPackGame> clusterPackGame = std::make_shared<ClusterPackGame>(config, streamCluster, cluster);
    clusterPackGame->initGame(config);
    clusterPackGame->startGame(config);
    return clusterPackGame;
}